web/js/tasks.js

```diff
@@ -882,34 +883,53 @@ Ext.onReady(function(){
         var content = document.getElementById('content');
         if( HEADER_HEIGHT + content.scrollHeight > window.innerHeight)
             window.scrollTo(0, taskPanel.getEl().getY());
+        // For a fresh task, we need to keep the status as draft, as we havent saved it yet!
+        Ext.getCmp('status_display').setText("Status: Draft");
     }
 
-    /* Add a callback to save tasks */
-    function saveTasks() {
-
-        // First we check if the time fields of all records are valid
+    // Validate the inputs in the task panel
+    function validateTasks() {
         var panels = tasksScrollArea.items;
-        var valids = true;
         for(var panel=0; panel<panels.getCount(); panel++) {
             if (!panels.get(panel).initTimeField.isValid() || !panels.get(panel).endTimeField.isValid()) {
-                valids = false;
-                break;
+                return false;
             }
         }
+        return true;
+    }
 
-        // If they are so, then we save the changes
-        if (valids) {
-            myStore.each(function(r) {
-                if (r.data['story'] != undefined)
-                    r.data['story'] = xmlencode(r.data['story']);
-                if (r.data['text'] != undefined)
-                    r.data['text'] = xmlencode(r.data['text']);
-            });
-            myStore.save();
+    // Add the new task to the taskstore
+    function addToMyStore() {
+        myStore.each(function(r) {
+            if (r.data['story'] != undefined)
+                r.data['story'] = xmlencode(r.data['story']);
+            if (r.data['text'] != undefined)
+                r.data['text'] = xmlencode(r.data['text']);
+        });
+        if (myStore.save()) {
+            return true;
+        }
+        return false;
+    }
+
+    /* Add a callback to save tasks */
+    function saveTasks() {
+        // First we check if the time fields of all records are valid, and then save
+        if (validateTasks()) {
+            addToMyStore();
         } else  // Otherwise, we print the error message
           App.setAlert(false, "Check For Invalid Field Values");
     }
 
+    // Implement autosave of data, when valid contents are typed in to the task fields
+    window.setInterval(function () {
+        if( unsavedChanges == true ) {
```

Comment:
Don't compare a boolean like this, you could have written `if(unsavedChanges)....`. In this case, you may even use `if(isUnsaved())`